File: src/main/factory/trafficSource/updateTrafficSourceFactory.ts
import { PrismaTrafficSourceRepository } from "../../../infra/repositories/trafficSource";
import { UpdateTrafficSourceUseCase } from "../../../app/useCases/trafficSource/updateTrafficSourceUseCase";
import { UpdateTrafficSourceController } from "../../../infra/controllers/trafficSource/updateTrafficSourceController";
import { UserGateway } from "../../../infra/gateways/user";

const prismaTrafficSourceRepository = new PrismaTrafficSourceRepository();
const userGateway = new UserGateway();

const updateTrafficSourceUseCase = new UpdateTrafficSourceUseCase(
  prismaTrafficSourceRepository,
  userGateway
);

const updateTrafficSourceController = new UpdateTrafficSourceController(
  updateTrafficSourceUseCase
);

const updateTrafficSource = {
  handle: updateTrafficSourceController.handle.bind(
    updateTrafficSourceController
  ),
};

export { updateTrafficSource };

File: internal/main/factory/trafficSource/updateTrafficSourceFactory.go
package trafficSourceFactories

import (
	trafficSourceUseCases "github.com/Lucas-Eduardo-Goncalves/arkyn-store/internal/app/useCases/trafficSource"
	"github.com/Lucas-Eduardo-Goncalves/arkyn-store/internal/domain/repositories"
	"github.com/Lucas-Eduardo-Goncalves/arkyn-store/internal/infra/adapters"
	trafficSourceControllers "github.com/Lucas-Eduardo-Goncalves/arkyn-store/internal/infra/controllers/trafficSource"
	contracts "github.com/Lucas-Eduardo-Goncalves/arkyn-store/internal/main/contracts"
)

func NewUpdateTrafficSourceFactory(trafficSourceRepository repositories.TrafficSourceRepository) contracts.Controller {
	updateTrafficSourceUseCase := trafficSourceUseCases.NewUpdateTrafficSourceUseCase(
		trafficSourceRepository,
		&adapters.DateGenerator{},
	)

	return trafficSourceControllers.NewUpdateTrafficSourceController(updateTrafficSourceUseCase)
}
